pkg/apis/helm.fluxcd.io/v1/types.go

```diff
@@ -89,6 +89,23 @@ type ValuesFromSource struct {
 	ChartFileRef *ChartFileSelector `json:"chartFileRef,omitempty"`
 }
 
+type NamespacedConfigMapKeySelector struct {
+	corev1.LocalObjectReference `json:",inline"`
```

Comment:
I don't think we should embed the `LocalObjectReference` since it could change in the future and it would conflict with the `Key` field. Also the `LocalObjectReference` has the `Name` field marked as optional but this would break Helm operator, a secret ref can't be resolved if the name is not specified. 

Instead of embedding  `LocalObjectReference` we could use
```suggestion
	Name string `json:"name"`
```

Reply:
I pretty much lifted the object definition from the earlier `corev1.ConfigMapKeySelector` and tried to modify it as little as possible. I don't recall the specifics of backwards compatibility regarding loading previous versions from storage when changing the types, but if that isn't an issue, then yes, definitely better to change it.